consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/ConsistencyChecker.java

```diff
@@ -202,10 +203,38 @@ public void run() {
         List<MigrationJob> jobs = new ArrayList<>();
         try {
             DataFindRequest findRequest = new DataFindRequest("migrationJob", migrationJobEntityVersion);
-            findRequest.where(and(
-                    withValue("configurationName = " + configuration.getConfigurationName()),
+            /*
+                and:
+                    whenAvailableDate <= new Date()
+                    or:
+                        jobExecutions# = 0
+                        not:
+                            jobExecutions.jobStatus $in ['COMPLETED_SUCCESS', 'COMPLETED_PARTIAL']
+                        and:
+                            not:
+                                jobExecutions.jobStatus $not_in [COMPLETED_SUCCESS', 'COMPLETED_PARTIAL']
+                            jobExecutions.actualStartDate < (new Date() - expectedExecutionMilliseconds)
+
+             */
+            findRequest.where(
+                and(
                     withValue("whenAvailableDate <= " + ClientConstants.getDateFormat().format(new Date())),
-                    not(withSubfield("jobExecutions", withValue("completedFlag = true")))));
+                    or(
+                        not(
+
+                            withSubfield("jobExecutions", withValue("jobStatus $in ['COMPLETED_SUCCESS', 'COMPLETED_PARTIAL']"))
+                        ),
+                        and(
+                            not(
+
+                                    withSubfield("jobExecutions", withValue("jobStatus $not_in ['COMPLETED_SUCCESS', 'COMPLETED_PARTIAL']"))
+                            ),
```

Comment:
Yeah I would assume that would happen. Not sure if it would be helpful, but this snippet of code was made in this discussion https://github.com/lightblue-platform/lightblue-migrator/pull/120#discussion_r26861317